ui/public/loadConfig.js

```diff
@@ -0,0 +1,20 @@
+// Load a json config.
+
+// Run loader as an anonymous function to protect the global namespace.
+// This function accesses a file "config.json" expected to be at the
+// root of the server and loads it into the global `GlobalReactConfig`.
+(function () {
+    var xobj = new XMLHttpRequest();
+    xobj.overrideMimeType("application/json");
+    // Run this in synchronous mode since the configuration should be loaded
+    // before the React app initializes.
+    xobj.open('GET', 'config.json', false);
+    xobj.onreadystatechange = function () {
+        if (xobj.readyState == 4 && xobj.status == "200") {
+            window.GlobalReactConfig = JSON.parse(xobj.responseText);
```

Comment:
Shouldn't there be some kind of callback triggered here to block bootstrapping of the app? What guarantee is there that config.json will be loaded by the time your react code runs?